src/main/kotlin/org/jetbrains/research/testspark/actions/TestSparkAction.kt

```diff
@@ -318,49 +325,79 @@ class TestSparkAction : AnAction() {
-                        )
-                    }
-                } else {
-                    // TODO: show the warning panel
+                if (codeTypeButtons[0].isSelected) {
+                    EvoSuite().generateTestsForClass(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
+                } else if (codeTypeButtons[1].isSelected) {
+                    EvoSuite().generateTestsForMethod(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
+                } else if (codeTypeButtons[2].isSelected) {
+                    EvoSuite().generateTestsForLine(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
                 }
             }
             visibilityController.isVisible = false
             dispose()
         }
 
-        private fun startEvoSuiteGeneration() = startUnitTestGenerationTool(tool = EvoSuite())
-        private fun startLLMGeneration() = startUnitTestGenerationTool(tool = Llm())
+        private fun startLLMGeneration() {
+            if (!testGenerationController.isGeneratorRunning(project)) {
+                val testSamplesCode = llmSampleSelectorFactory.getTestSamplesCode()
+
+                if (codeTypeButtons[0].isSelected) {
+                    Llm().generateTestsForClass(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
+                } else if (codeTypeButtons[1].isSelected) {
+                    Llm().generateTestsForMethod(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
+                } else if (codeTypeButtons[2].isSelected) {
+                    Llm().generateTestsForLine(
+                        project,
+                        psiHelper,
+                        caretOffset,
+                        fileUrl,
+                        testSamplesCode,
+                        testGenerationController,
+                    )
+                }
+            }
+            visibilityController.isVisible = false
+            dispose()
+        }
```

Comment:
The same duplication got introduced again. Maybe, you have not rebased on the `development`?